src/rest-server/src/models/token.js

```diff
@@ -49,58 +65,83 @@ const sign = async (username, application, expiration) => {
  * @param {Object} data - id -> token format data object (data stored in k8s secret)
  * @returns {Object} Purged data in the same format
  */
-const purge = (data) => {
+const purge = async (data, jobSpecific) => {
   const result = {};
   for (const [key, val] of Object.entries(data)) {
     try {
+      // expired tokens will be removed
       jwt.verify(val, secret);
-      result[key] = val;
     } catch (err) {
-      // pass;
+      continue;
+    }
+    if (jobSpecific) {
```

Comment:
If the user submitted a lot of jobs, e.g. > 1000, this will cause 1000+ rpc call to database, which will be a huge time delay. I suggest to collect all keys in `data` first, then validate them in db one time. You can use [this syntax](https://sequelize.org/master/manual/model-querying-basics.html#shorthand-syntax-for--code-op-in--code-) to select multiple frameworks' states.

Reply:
> If the user submitted a lot of jobs, e.g. > 1000, this will cause 1000+ rpc call to database, which will be a huge time delay. I suggest to collect all keys in `data` first, then validate them in db in one time. You can use [this syntax](https://sequelize.org/master/manual/model-querying-basics.html#shorthand-syntax-for--code-op-in--code-) to select multiple frameworks' states.

Fixed, thx